commons/src/main/java/com/powsybl/commons/extensions/ExtensionAdderProviders.java

```diff
@@ -0,0 +1,73 @@
+package com.powsybl.commons.extensions;
+
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Suppliers;
+import com.powsybl.commons.PowsyblException;
+
+/**
+ * A utility class to help finding providers using ServiceLoader.
+ *
+ * @author Jon Harper <jon.harper at rte-france.com>
+ *
+ */
+//Don't bother with generics because serviceloader doesn't return them
+//and we put them in a cache where we can't propagate the generic types.
+@SuppressWarnings("rawtypes")
+final class ExtensionAdderProviders {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExtensionAdderProviders.class);
+
+    private ExtensionAdderProviders() {
+    }
```

Comment:
Sonar should warn, but static variables should be declared before this constructor

Reply:
done